node/chainSimulator/components/coreComponents_test.go

```diff
@@ -0,0 +1,269 @@
+				Type: "blake2b",
+			},
+			TxSignHasher: config.TypeConfig{
+				Type: "blake2b",
+			},
+			AddressPubkeyConverter: config.PubkeyConfig{
+				Length: 32,
+				Type:   "hex",
+			},
+			ValidatorPubkeyConverter: config.PubkeyConfig{
+				Length: 128,
+				Type:   "hex",
+			},
+			GeneralSettings: config.GeneralSettingsConfig{
+				ChainID:               "T",
+				MinTransactionVersion: 1,
+			},
+			Hardfork: config.HardforkConfig{
+				PublicKeyToListenFrom: "41378f754e2c7b2745208c3ed21b151d297acdc84c3aca00b9e292cf28ec2d444771070157ea7760ed83c26f4fed387d0077e00b563a95825dac2cbc349fc0025ccf774e37b0a98ad9724d30e90f8c29b4091ccb738ed9ffc0573df776ee9ea30b3c038b55e532760ea4a8f152f2a52848020e5cee1cc537f2c2323399723081",
+			},
+		},
+		EnableEpochsConfig: config.EnableEpochs{},
+		RoundsConfig: config.RoundConfig{
+			RoundActivations: map[string]config.ActivationRoundByName{
+				"DisableAsyncCallV1": {
+					Round: "18446744073709551615",
+				},
+			},
+		},
+		EconomicsConfig: config.EconomicsConfig{
+			GlobalSettings: config.GlobalSettings{
+				GenesisTotalSupply: "2000000000000000000000",
+				MinimumInflation:   0,
+				YearSettings: []*config.YearSetting{
+					{
+						Year:             0,
+						MaximumInflation: 0.01,
+					},
+				},
+			},
+			FeeSettings: config.FeeSettings{
+				GasLimitSettings: []config.GasLimitSetting{
+					{
+						MaxGasLimitPerBlock:         "10000000000",
+						MaxGasLimitPerMiniBlock:     "10000000000",
+						MaxGasLimitPerMetaBlock:     "10000000000",
+						MaxGasLimitPerMetaMiniBlock: "10000000000",
+						MaxGasLimitPerTx:            "10000000000",
+						MinGasLimit:                 "10",
+						ExtraGasLimitGuardedTx:      "50000",
+					},
+				},
+				GasPriceModifier:       0.01,
+				MinGasPrice:            "100",
+				GasPerDataByte:         "1",
+				MaxGasPriceSetGuardian: "100",
+			},
+			RewardsSettings: config.RewardsSettings{
+				RewardsConfigByEpoch: []config.EpochRewardSettings{
+					{
+						LeaderPercentage:                 0.1,
+						DeveloperPercentage:              0.1,
+						ProtocolSustainabilityPercentage: 0.1,
+						ProtocolSustainabilityAddress:    "erd1932eft30w753xyvme8d49qejgkjc09n5e49w4mwdjtm0neld797su0dlxp",
+						TopUpGradientPoint:               "300000000000000000000",
+						TopUpFactor:                      0.25,
+						EpochEnable:                      0,
+					},
+				},
+			},
+		},
+		ChanStopNodeProcess: make(chan endProcess.ArgEndProcess),
+		InitialRound:        0,
+		NodesSetupPath:      "../../../sharding/mock/testdata/nodesSetupMock.json",
+		GasScheduleFilename: "../../../cmd/node/config/gasSchedules/gasScheduleV7.toml",
+		NumShards:           3,
+		WorkingDir:          ".",
+		MinNodesPerShard:    1,
+		MinNodesMeta:        1,
+	}
+}
+
+func TestCreateCoreComponents(t *testing.T) {
+	t.Parallel()
+
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		comp, err := CreateCoreComponents(createArgsCoreComponentsHolder())
```

Comment:
should we call comp.Close() at the end of this test?

Reply:
nope, added for coverage only